---
layout: post
title: Configuration of TopHat, Bowtie, Samtools and Cufflinks on Ubuntu
categories: [bioinformatics, NGS, ubuntu]
tags: [Tophat, Bowtie, Samtools, Cufflinks, configure]
---

Once you had removed all adapter sequences from your FASTQ files, you were ready to map them to the corresponding reference genome. While tools like BLAST and BLAT are
powerful methods, they are not specialized for the vast amount of data generated by next-generation sequencing. So this week I used a next-generation specific read alignment programs, such Tophat, Bowtie, Samtools and Cufflinks below.

### What are Tophat, Bowtie, Samtools and Cufflinks?

#### [TopHat](http://tophat.cbcb.umd.edu/) is a program that aligns RNA-Seq reads to a genome in order to identify exon-exon splice junctions. It is built on the ultrafast short read mapping program Bowtie. TopHat runs on Linux and OS X.

#### [Bowtie](http://bowtie-bio.sourceforge.net/index.shtml) is an ultrafast, memory-efficient short read aligner geared toward quickly aligning large sets of short DNA sequences (reads) to large genomes. It aligns 35-base-pair reads to the human genome at a rate of 25 million reads per hour on a typical workstation. Bowtie indexes the genome with a Burrows-Wheeler index to keep its memory footprint small: for the human genome, the index is typically about 2.2 GB (for unpaired alignment) or 2.9 GB (for paired-end or colorspace alignment).

#### [Samtools](http://www.biomedcentral.com/content/supplementary/1471-2105-13-42-s1/Cloud-BioLinux-Package-Documentation/docs/samtools.html) provide various utilities for manipulating alignments in the SAM format, including sorting, merging, indexing and generating alignments in a per-position format.

#### [Cufflinks](http://cufflinks.cbcb.umd.edu/) assembles transcripts, estimates their abundances, and tests for differential expression and regulation in RNA-Seq samples. It accepts aligned RNA-Seq reads and assembles the alignments into a parsimonious set of transcripts. Cufflinks then estimates the relative abundances of these transcripts based on how many reads support each one, taking into account biases in library preparation protocols.

### The Workflow of Using Tophat, Bowtie, Samtools And Cufflinks:

- Step 1: Download the reference genome and **Tophat**, **Bowtie**, **Samtools** and **Cufflinks**.
- Step 2: Configure the working environment for **Tophat**, **Bowtie**, **Samtools**  and **Cufflinks**.
- Step 3: Build index for reference genome with **Bowtie**.
- Step 3: Align paired-end RNA-seq reads to reference genome via **Tophat**.
- Step 4: Use **Cufflinks** to calculate transcript abundances (**FPKM**) in RNA-seq samples
- Step 5: Use **R** to calculate the correlation of **FPKM** between the both replications.
- Step 6: Merge the replications and run **Tophat** and **Cufflinks** again.

#### Download the zebrafish reference genome
The next two weeks, all my works have been showed in the workflow above. In term of the paper ([Systematic identification of long noncoding RNAs expressed during zebrafish embryogenesis](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3290793/)) I was working, firstly I should figure out where the zebrafish reference genome the paper used is from. And it told me that:

![](http://i.imgur.com/0j2PDa5.png)

From this paragraph in the paper, it is supereasy to search out zebrafish reference genome from
following procedure:

```
The keyword typing into Google search bar: zebrafish reference genome -> Ensemble ->
Download DNA sequence(FASTA) -> ftp://ftp.ensembl.org/pub/release-71/fasta/danio_rerio/dna/
```
Note: The alternative link: ( ftp://ftp.sanger.ac.uk/pub/zfish/assembly/Zv9/ )

### Configure the working environment for Tophat, Bowtie, Samtools And Cufflinks

If you read [**Getting Started**](http://tophat.cbcb.umd.edu/tutorial.shtml) of **TopHat**, you could know that **Tophat** depends on **Bowtie**, **Samtools** and **Boost**. So you need to install **Bowtie**, **Samtools** and **Boost** before the Tophat could work smoothly on your PC. Follow the **Getting Started**, the smplified the procedures I have done is below. (If you want to know everything with very detials, you could click the **Getting Started** of **Tophat above**)

#### Install Bowtie

First, go to your home directory:
```
$ cd ~/
```
#### Download the binary file and decompress it:

```
$ wget http://sourceforge.net/projects/bowtie-bio/files/bowtie/0.12.7/bowtie-0.12.7-linux-x86_64.zip
$ tar xvfz bowtie-0.12.7-linux-x86_64.zip
$ cd bowtie-0.12.7
```
#### Set the environment variables of Bowite

Copy its executable files to /usr/bin:

```
$ sudo cp bowtie /usr/bin
$ sudo cp bowtie-build /usr/bin
$ sudo cp bowtie-inspect /usr/bin
```
or use the method I have metioned at previous journal ([Configure the Path And Set SSH with Tricks (Ubuntu)](http://lushen.github.com/en/2013/08/SSH-2013/)).

#### Install Samtools

Download the source file and decompress it:

```
$ cd ~/
$ wget http://sourceforge.net/projects/samtools/files/samtools/0.1.16/samtools-0.1.16.tar.bz2
$ tar xvfj samtools-0.1.16.tar.bz2
```
> Note: `tar xvfj` command is used to uncompress file in `bz2` format. Use `tar xvfz` instead to uncompress file in `gzip` format.

#### Complie Samtools:

```
$ cd samtools-0.1.16
$ make
```
#### Set the path of Samtools as same as Bowtie:

Just see the procedures above.

#### Install Tophat

Download the binary file and untar it:

```
$ cd ~/
$ wget http://tophat.cbcb.umd.edu/downloads/tophat-1.3.0.Linux_x86_64.tar.gz
$ tar xvfz tophat-1.3.0.Linux_x86_64.tar.gz
$ cd tophat-1.3.0.Linux_x86_64
```
#### Set the path of Tophat as same as Bowtie

Just see the procedures above.

#### Use the command line to check if the Tophat could work

```
$ wget http://tophat.cbcb.umd.edu/test_data.tar.gz
$ tar zxvf test_data.tar.gz
$ cd test_data
$ tophat -r 20 test_ref reads_1.fq reads_2.fq
```
if the result like that, it could work successfully.

![](http://i.imgur.com/C6F80Sf.png)

The process of configuring **Cufflinks** is as same as **Tophat**, **Bowtie** and **Samtools**, thus I didn't repeat again here. By the way, **Cufflinks** homepage provide very detialed [tutorial](http://cufflinks.cbcb.umd.edu/tutorial.html) to get started.

#### Note: The guide does not provide a ready-made test data, you could make these by yourself as followings:

- Click "test data" -> find the web page that show the all data in it -> rigth-click and choose "select all" -> right-click and copy it.

- Build new file named "test_data.sam" -> open it with **vim** and paste the data into and save it -> you can use it with command line direccly:

```
$ cufflinks ./test_data.sam
```
And you could see as below

![](http://i.imgur.com/5XrGWQa.png)

After these, we have finished all the processes of setting the environment for **Tophat**, **Bowtie**, **Samtools** and **Cufflinks**. The next journal I will provide more detials about how to apply them at practice. 
