---
layout: post
title: Pre-processing And Quality Control of the Raw NGS Data I:Convert SRA Format to FASTQ Format
categories: [reference, basic knowledge]
tags: [courses, bioinformatics]
---

## The Workflow of Pre-processing And Controlling the Raw NGS Data:  
- Step 1: Download the SRA files And Install Related Software 
- Step 2: Convert SRA Format to FASTQ Format 
- Step 3: Filter the Lower Quality Data 

![](http://4.bp.blogspot.com/-J3t4zTqvPkU/UYX33zWb1NI/AAAAAAAADXk/al-mf09Ysq0/s1600/Screen%2BShot%2B2013-05-04%2Bat%2B11.07.53%2BPM.png)

## What is SRA and FASTQ format?
- [SRA](http://en.wikipedia.org/wiki/Short_Read_Archive)(Sequence Read Archive): The Sequence Read Archive originally Short Read Archive is a
bioinformatics database that provides a public repository for DNA sequencing data, especially the
"short reads" generated by High-throughput sequencing, which are typically less than 1000 base
pairs in length.
- [FASTQ](http://en.wikipedia.org/wiki/Fastq) format: is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores.

Based on the first paper ([Systematic identification of long noncoding RNAs expressed during zebrafish embryogenesis](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3290793/)) I worked, the RNA-seq and Chip-seq data are both SRA formats. Because SAR format data cannot be used directly, before analysing these data in next step, I should
convert that into FASTQ format.

## Download the SRA files And Install Related Softwares 
The steps of downloading SRA files you could reference the journal I wrote before: [The Fast Download with FileZilla (Ubuntu)](http://lushen.github.com/en/2013/05/FileZilla-2013/). After preparing all the SRAfiles, nextstepis that installing specific softwares to process data and then improve the quality of your data. I choose the SRAtoolkit and FASTXtoolkit to do that, which could help you to pre-process your files and filter the data with a low quality respectively.

- Currently, there are countless softwares that could be used to convert SRA to FASTQ format, here I just used **SRA toolkit**.
Here list the procedure to download it(eg. CentOS Linux 64 bit architecture ):  

```
$ wget "http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-centos_linux64.tar.gz"  
$ tar xzf sratoolkit.current-centos_linux64.tar.gz  
```
- Open the directory that has untarred and find ```fastq-dump``` in it.  

```
$ cd sratoolkit.2.3.2-5-centos_linux64/bin/  
$ ls  
```

## Convert SRA Format to FASTQ Format  

Using ```fastq-dump``` command to convert SRA format to fastq format:
```
$ /home/shenlu/temp/sratoolkit.2.3.2-centos_linux64/bin/fastq-dump /home/hchou/data2/fish/Downloads/ERR034124.sra
```
After that, you could see the FASTQ files converted from SRA in the directory that stored your
SRA files (such as '/home/hchou/data2/fish/Downloads/ERR034124.sra' above ).
> Note: 
> * if you want use sratoolkit, the prerequisite is that your computer must be 64-bit. Otherwise, sratoolkit cannot work.  
> * If you want use ' fastq-dump ' or others in other directories, you should claim the position of all the path of the original fastq-dump or others(like '/home/shenlu/temp/sratoolkit.2.3.2-centos_linux64/bin/fastq-dump' command line above) and the directory now you opened is the position where the converted file will stay at.  
> * You also could use:`/home/hchou/data2/fish/Downloads/*.sra` to convert all of the file with the end of .sra , but the speed of this method is a little bit lower that just converting one each time. But sometimes this way is very conveneint.  
> * opening the file of .fastq, the command is these: less, more, head and tail and vim.  
> * The command of multipling files is that: `cp`, `scp` (that can control the long-distance file), `rm` and `mv`.  
> * You do not have to know the path of the file, you just drag the file that you want to use in the terminal and you could see the path of this file.  
> * Due to the parameter of data from this paper SOLID is color space, when using `fastq-dump` command, adding -B in the front of the file path (-C convert dumpbase to dumpcs).  


## Filter the Lower Quality Data

