---
layout: post
title: How to Use TACC (Texas Advanced Computing Center) to Facilitate Data Analysis (Ubuntu) 
categories: [Ubuntu]
tags: [ssh, configure, supercomputer]
---

![](http://i.imgur.com/PSNKgUT.jpg)

Supercomputer is one of the essential tools for bioinformatics, especially encountering the big data,
which will dramatically speed up your work process. [TACC](https://www.tacc.utexas.edu/)(Texas Advanced Computing Center) at the University of Texas at Austin, United States, is a research center for advanced computational
science, engineering and technology. This summer I registered an account at TACC and [XSEDE](https://www.xsede.org/)(The Extreme Science andEngineering Discovery Environment)

![](http://i.imgur.com/oVRzY9j.gif)

and used it involving these aspects:

- 1.Login in TACC with Terminal.
- 2.Load Modules And Transfer Files or Directories in It from Remote Distance.
- 3.Write Job Scripts And Submit It to Run 

## How to Login TACC
After registered your account, the most regulator approach to login supercomputer with **SSH**.

```
$ ssh lushen@stampede.tacc.utexas.edu
```
[Here](http://lushen.github.com/en/2013/08/SSH-2013/) is a more easier way to login it. (I mentioned at last journal)

After you login, you should see something like that:

![](http://i.imgur.com/j3aRebE.png)

### How to Load Bowtie, Tophat and Samtools at TACC
After logging into TACC, you should load the softwares you want to use

```
$ moudle load softwareName(eg. bowtie)
```
> Note: You should load Bowtie first and Tophat afterwards.
And If you have loaded these software but you close your supercomputer once. These software will disappear at the end. So you need to use the command: module save, which will set these softwares are default and never disappear unless you want to change that.

After that, you could remotely transfer the files or directories use the command (if you want see the detials, use the command line: 
```$ man scp (or $ scp --help)```.

```
$ scp -r /work/fish/shenlu/2-4_cell_2
lushen@stampede.tacc.utexas.edu:/scratch/02473/lushen/temp
```
## How to Write job scripts And Submit it to Clusters
### 1. Edit a job scripts (the more details: [Stampede User Guide](https://www.tacc.utexas.edu/user-services/user-guides/stampede-user-guide))

{% highlight r %}
#!/bin/bash
#SBATCH -J myjob1
# job name
#SBATCH -o myjob1.o%j
# output and error file name (%j expands to jobID)
#SBATCH -n 32
# total number of mpi tasks requested
#SBATCH -p normal # queue (partition) -- normal, development, etc.
#SBATCH -t 24:00:00
# run time (hh:mm:ss) - 24.0 hours
#SBATCH --mail-type=end # email me when the job finishes
#The statements below are the commands that you want supercomputer to implement.
tophat -p 16 --library-type fr-unstranded /scratch/02473/lushen/temp/bowtie/zebrafish_index
/scratch/02473/lushen/temp/2-4_cell/2-4_cell_1.fastq /scratch/02473/lushen/temp/2-4_cell/2-
4_cell_2.fastq -G /scratch/02473/lushen/temp/zebrafish_genome/Danio_rerio_annotation
{% endlight %}

### 2. Use 'sbatch' to submit the command line in the cluster of supercomputer

```
$ sbatch job_script_1.txt job_script_2.txt
```
### 3. Show the job that have submitted to check if your edited scripts are submitted

```
$ showq -u
```

### 4. Cancel your job you have submitted

```
$ scancel 870919 870922
```
[User guides](https://www.tacc.utexas.edu/user-services/user-guides)




