---
layout: post
title: Pre-processing And Quality Control of the Raw NGS Data I:Convert SRA Format to FASTQ Format
categories: [reference, basic knowledge]
tags: [courses, bioinformatics]
---

## What is SRA and FASTQ format?
- [SRA](http://en.wikipedia.org/wiki/Short_Read_Archive)(Sequence Read Archive): The Sequence Read Archive originally Short Read Archive is a
bioinformatics database that provides a public repository for DNA sequencing data, especially the
"short reads" generated by High-throughput sequencing, which are typically less than 1000 base
pairs in length.
- [FASTQ](http://en.wikipedia.org/wiki/Fastq) format: is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores.

![](http://4.bp.blogspot.com/-J3t4zTqvPkU/UYX33zWb1NI/AAAAAAAADXk/al-mf09Ysq0/s1600/Screen%2BShot%2B2013-05-04%2Bat%2B11.07.53%2BPM.png)

## The Workflow of Pre-processing And Controlling the Raw NGS Data:  
- Step 1: Download the SRA files And Install Related Software 
- Step 2: Convert SRA Format to FASTQ Format 
- Step 3: Filter the Lower Quality Data 

Based on the first paper ([Systematic identification of long noncoding RNAs expressed during zebrafish embryogenesis](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3290793/)) I worked, the RNA-seq and Chip-seq data are both SRA formats. Because SAR format data cannot be used directly, before analysing these data in next step, I should
convert that into FASTQ format.

## Download the SRA files And Install Related Softwares 
The steps of downloading SRA files you could reference the journal I wrote before: [The Fast Download with FileZilla (Ubuntu)](http://lushen.github.com/en/2013/05/FileZilla-2013/). After preparing all the SRAfiles, nextstepis that installing specific softwares to process data and then improve the quality of your data. I choose the SRAtoolkit and FASTXtoolkit to do that, which could help you to pre-process your files and filter the data with a low quality respectively.

- Currently, there are countless softwares that could be used to convert SRA to FASTQ format, here I just used **SRA toolkit**.
Here list the procedure to download it(eg. CentOS Linux 64 bit architecture ):  

```
$ wget "http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-centos_linux64.tar.gz"  
$ tar xzf sratoolkit.current-centos_linux64.tar.gz  
```
- Open the directory that has untarred and find ```fastq-dump``` in it.  

```
$ cd sratoolkit.2.3.2-5-centos_linux64/bin/  
$ ls  
```

## Convert SRA Format to FASTQ Format  

- Using ```fastq-dump``` command to convert SRA format to fastq format:
```
$ /home/shenlu/temp/sratoolkit.2.3.2-centos_linux64/bin/fastq-dump /home/hchou/data2/fish/Downloads/ERR034124.sra
```
After that, you could see the FASTQ files converted from SRA in the directory that stored your
SRA files (such as '/home/hchou/data2/fish/Downloads/ERR034124.sra' above ).  

> Note:  
> * if you want use sratoolkit, the prerequisite is that your computer must be 64-bit. Otherwise, `sratoolkit` cannot work.  
> * If you want use `fastq-dump` or others in other directories, you should claim the position of all the path of the original `fastq-dump` or others (like `/home/shenlu/temp/sratoolkit.2.3.2-centos_linux64/bin/fastq-dump` command line above) and the directory now you opened is the position where the converted file will stay at.  
> * You also could use:`/home/hchou/data2/fish/Downloads/*.sra` to convert all of the file with the end of `.sra` , but the speed of this method is a little bit lower that just converting one each time. But sometimes this way is very conveneint.  
> * opening the file of `.fastq`, the command is these: `less`, `more`, `head` and `tail` and `vim`.  
> * The command of multipling files is that: `cp`, `scp` (that can control the long-distance file), `rm` and `mv`.  
> * You do not have to know the path of the file, you just drag the file that you want to use in the terminal and you could see the path of this file.  
> * Due to the parameter of data from this paper `SOLID` is `color space`, when using `fastq-dump` command, adding `-B` in the front of the file path (`-C` convert dumpbase to dumpcs).  

- Calculate out corresponding quality sore value and save these in text files and show these with png or jpg files

After that, you should executed the command ```fastx_quality_stats``` to calculate out nucleotides
distribution (there is no quailty information) and output a text file stored these information.  

```  
$ /home/shenlu/fastx_bin/bin/fastx_quality_stats -i SRR372791_1.fastq -o SRR372791_1.txt  
```  

, it came out an error that ```fastx_quality_stats```: Invalid quality score value(...) '. That you should
insert a ```-Q33``` argument into the command line and run that.   

```   
$ /home/shenlu/fastx_bin/bin/fastx_quality_stats -i SRR372791_1.fastq -o SRR372791_1.txt -Q33  
```  
and it will work normally. If you get the primary statistical information about your data, you should first check out if the quality of the data match for your needs (you can rename the FASTQ and text files according to the paper to make them more understandable, like name ```SRR372791_1.fastq``` and ```SRR372791_1.txt``` to
```2-4_cell_1.fastq``` and ```2-4_cell_1.txt```).   

```   
$ fastq_quality_boxplot_graph.sh -i SRR372791_1.txt -o 2-4_cell_87_1.png -t “2-4 cell_1 ” | fastq_quality_boxplot_graph.sh -i SRR372791_2.txt -o 2-4_cell_87_2.png -t “2-4 cell_2 ”   
```   

> Note: Using `|` to divide your command lines is to avoid you to repeat typing the same or similar
command each time and make you just get done everything once.

After that you would see the results like that below: 

**2-4_cell_87_1.png**

![](http://i.imgur.com/XSdrsa1.png)

**2-4_cell_87_2.png**

![](http://i.imgur.com/7zwEkyh.png)

Background Knowledge: [Phred quality score](http://en.wikipedia.org/wiki/Phred_quality_score)  

According to the table above, we could see from the **2-4_cell_87_1.png**, from the 50 bp, The
quality of the sequencing reduce sharply. And even if we have filtered the low quality reads, it is
still hard to get a desirable results (such as the picture below, which has been treated by the
command ```FASTQ Quality Filter``` from FASTX toolkit with the argument: ```-q 20 -p 80```, which
means the 80 percent of bases has the accuracy more than 99% or at least 20 Phred quality score).

![](http://i.imgur.com/ySSwhaV.png)

In this case, before we used the first fastq file, we should trim a part of length with low quality score
before we executed any command to filter low-quality score reads. So we should use ```fastx_trimmer``` to do that 

```  
$ fastx_trimmer -f 1 -l 50 -i SRR372787_1.fastq -o SRR372787_1_trimmer.fastq  
```   

which means keeping the bases from 1st to 50th. After that we could use [Trimmomatic](http://www.usadellab.org/cms/index.php?page=trimmomatic) to control the quality of the data in both fasq files.  


## Filter the Lower Quality Data

[Trimmomatic](http://www.usadellab.org/cms/index.php?page=trimmomatic)(A flexible read trimming tool for Illumina NGS data) performs a variety of useful trimming tasks for illumina paired-end and single ended data.The selection of trimming steps and their associated parameters are supplied on the command line.
### How to use Trimmomatic 0.30  
After downloading it, we should just use it as the manual mentioned that offered by its homepage. 

```  
$ java -classpath /home/shenlu/temp/Trimmomatic-0.30.jar org.usadellab.trimmomatic.TrimmomaticPE -phred33 SRR372787_1_trimmer.fastq SRR372787_2.fastq SRR372787_1_paired.fastq SRR372787_1_unpaired.fastq SRR372787_2_paired.fastq SRR372787_2_unpaired.fastq ILLUMINACLIP:/work/fish/ILadaptors_truseq.fa:2:40:15 LEADING:20 TRAILING:20 SLIDINGWINDOW:4:20 MINLEN 36  
```  

The actual meaning of these arguments above is that:  
- **ILLUMINACLIP**: Remove adapters  
- **LEADING**: Remove leading low quality or N bases (below quality 20)  
- **TRAILING**: Remove trailing low quality or N bases (below quality 20)  
- **SLIDINGWINDOW**: Scan the read with a 4-base wide sliding window, cutting when the average quality per base drops below 20  
- **MINLEN**: Drop reads below the 36 bases long and the more details about arguments in Trimmomatic.  


